digdag-plugin-utils/src/main/java/io/digdag/util/BaseOperator.java

```diff
@@ -16,17 +16,17 @@
 public abstract class BaseOperator
         implements Operator
 {
-    protected final Path workspacePath;
+    protected final Path projectPath;
     protected final Workspace workspace;
     protected final TaskRequest request;
 
     protected final List<Config> inputs;
     protected final List<Config> outputs;
 
-    public BaseOperator(Path workspacePath, TaskRequest request)
+    public BaseOperator(Path projectPath, TaskRequest request)
     {
-        this.workspacePath = workspacePath;
-        this.workspace = new Workspace(workspacePath);
+        this.projectPath = projectPath;
+        this.workspace = Workspace.ofTaskRequest(projectPath, request);
```

Comment:
[minor] This variable is closed in `run()` not `close()` or something explicitly. So it seems to be a bit confusing life time to me. How about removing this variable and modifying `run()` like this?
```
    @Override
    public TaskResult run(TaskExecutionContext ctx)
    {
        RetryControl retry = RetryControl.prepare(request.getConfig(), request.getLastStateParams(), false);
        try {
            try (Workspace workspace = Workspace.ofTaskRequest(projectPath, request)) {
                return runTask(ctx, workspace);
            }
        }
```